integrations/electricity/linky.js

```diff
@@ -118,82 +42,64 @@ function disconnect() {
-    .query(query)
-    .type('form')
-    // Those are required to avoid the client to cache responses
-    .set('Cache-Control', 'no-cache')
-    .set('If-None-Match', '*')
-    .send(payload);
-
-  const responseURL = getResponseURL(res);
-
-  // Check if the response URL is a redirect to the login page
-  if (responseURL && responseURL.includes('Login')) {
-    throw new Error('We\'re supposed to be logged in at this stage');
+async function fetchActivities(usagePointId, frequency, startDate, endDate, logger) {
+  let url;
+  if (frequency === 'hour') {
+    url = '/v3/metering_data/consumption_load_curve';
+  } else if (frequency === 'day') {
+    url = '/v3/metering_data/daily_consumption';
   }
+  logger.logDebug(`Fetching at frequency=${frequency}, startDate=${startDate}, endDate=${endDate}`);
+  const res = await manager.fetch(
+    `${url}?usage_point_id=${usagePointId}&start=${startDate}&end=${endDate}`,
+    {},
+    logger
+  );
 
-  const json = JSON.parse(res.text);
-  if (json.etat.valeur === 'erreur') {
-    throw new Error(`Error while fetching data. More info: ${JSON.stringify(json)}`);
-  }
-  if (json.etat.valeur === 'nonActive') {
-    if (frequency === 'hour') {
-      return fetchActivities('day', startDate, endDate, logger);
+  if (!res.ok) {
+    if (res.status === 403) {
+      throw new HTTPError(res.headers.get('www-authenticate'), res.status);
     }
-    throw new Error(`No available data for the selected period. More info: ${JSON.stringify(json)}`);
-  }
-
-  const { data, periode } = json.graphe;
-  const offset = json.graphe.decalage || 0;
-
-  if (!Array.isArray(data)) {
-    throw new Error(`Unexpected data: ${JSON.stringify(json)}`);
+    if (res.status === 404) {
+      // no data for this point
+      if (frequency === 'hour') {
+        // Try with 'day'
+        logger.logDebug('Couldn\'t access hourly data. Trying with daily..');
+        return fetchActivities(usagePointId, 'day', startDate, endDate, logger);
+      }
+    }
+    throw new HTTPError(await res.text(), res.status);
   }
 
-  // TODO: Double check timezones?
-  const startMoment = moment(periode.dateDebut, 'DD/MM/YYYY');
-  const endMoment = moment(periode.dateFin, 'DD/MM/YYYY');
-
-  // Taken from https://github.com/bokub/linky/blob/master/index.js#L156-L183
-  data.splice(0, offset);
-  data.splice(-offset, offset);
-
-  /*
-    Schema:
-    data = [
-      {"ordre": 1066, "valeur": 2.206}
-      ...
-    ]
-    `ordre` is an index
-    `valeur` is in kWh
-    Negative `valeur` means data is unknown
-  */
-
-  const parseValue = d => (d.valeur >= 0 ? d.valeur * 1000 : 0);
+  const json = (await res.json());
+
+  const data = json.usage_point
+    .find(d => d.meter_reading.usage_point_id === usagePointId)
+    .meter_reading;
+  const intervalLengthSeconds = data.reading_type.interval_length;
+  const unit = data.reading_type.unit;
+  logger.logDebug(data);
+
+  // TODO: This should be parsed in French timezone instead of local?
```

Comment:
Is this something that is considered breaking, or are you fine with leaving it like this at this for now?

Reply:
We should change it now I think. Problem happens when local timezone != french timezone, e.g. when the phone is abroad.
Do you know the easiest way to parse a datetime with a given timezone?

Reply:
Hmm, there's [moment-timezone](https://momentjs.com/timezone), but might be a bit too much for this use case. I guess we could just add a utc-offset to the date, so that it's always evaluated as the correct timezone?

Haven't tested it, but I assume something like this could work:
```	
  const FRENCH_UTC_OFFSET = '+01:00';
  const startInFrenchTimezone =	data.start + FRENCH_UTC_OFFSET;
  const startMoment = moment(startInFrenchTimezone, 'YYYY-MM-DDZ');
```

https://momentjs.com/docs/#/parsing/string-format/

Reply:
The problem is that the french offset is not fixed (daylight saving etc..)

Reply:
That one can be fixed with `moment(data.start).isDST()`, but if we might need this later for other timezones, it might be better to use `moment.tz()` with the moment-timezone plugin :)

Reply:
absolutely. Done now